src/conversions.rs

```diff
@@ -508,7 +509,10 @@ impl<C: Clone, T: FromJSValConvertible<Config=C>> FromJSValConvertible for Vec<T
 
             for i in 0..length {
                 let mut val = RootedValue::new(cx, UndefinedValue());
-                assert!(JS_GetElement(cx, obj.handle(), i, val.handle_mut()));
+                if !JS_GetElement(cx, obj.handle(), i, val.handle_mut()) {
+                    // On JS exception or error exit loop
```

Comment:
As far as I can tell, JS_GetElement is an external C function, and it doesn't use any of the `throw_js_error` stuff that we have written in Rust.